src/visumorph/scale.py

```diff
@@ -40,20 +40,16 @@ def scale(image, scale):
     >>> scaled_img = vm.scale(img, 1.15)
     """
 
-    if not isinstance(image, Image):
-        raise TypeError("The provided image must be a valid VisuMorph Image "
-                        "object.")
+    if not isinstance(image, visumorph.Image):
+        raise TypeError("Input 'image' must be a valid VisuMorph Image.")
+    if not isinstance(scale, (int, float)):
+        raise TypeError("Input 'scale' must be a number.")
 
-    if not type(scale) in (float, int):
-        raise TypeError("The scale provided must be a number.")
+    # Calculate new dimensions
+    new_width = round(image.width * scale)
+    new_height = round(image.height * scale)
 
-    if scale <= 0:
-        raise ValueError("The scale provided must be a positive, non-zero "
-                         "number.")
+    # Scale the image
+    scaled_image = image.resize((new_width, new_height))
```

Comment:
We do not have `.resize()` method in our `Image` object. Have you ran the tests?

Reply:
thanks, changes made accordingly

Reply:
Resolving this as this line is removed.